### PR TITLE
cmd/contour: extract getDAGBuilder helper function

### PR DESCRIPTION
**Pull request by skriss**
Extracts all logic in doServe() related to constructing
the dag.Builder into a helper function.

Updates #3453.

Signed-off-by: Steve Kriss <krisss@vmware.com>

### CODE REVIEW

**Comment by skriss** [comment]
@projectcontour/maintainers I'm not 100% sure yet where this refactoring is going to end up, but as a starting point I'm trying to extract some atomic chunks of logic from the `doServe()` function into their own helpers, and to start to add some tests. Feel free to provide feedback on if you think there's a better approach.

**Review by youngnick — APPROVED** [review]
LGTM, anything we can do to make that functions smaller is good in my book.

**Review by sunjayBhatia — APPROVED** [review]
LGTM, nice start to getting more test coverage

Maybe something to consider for the future would be mutator methods on `dag.Builder`?
So we could do something like `dag.NewBuilder().WithFoo(...)`
Not sure if we like that pattern and have used it in Contour or want to just use a more structured `New` function instead so we can test in the `dag.Builder` tests?
Something needs to ensure `serve.go` is correctly wiring things up and this is a good start

File: cmd/contour/serve_test.go
Comment (stevesloka) [inline cmd/contour/serve_test.go:66]: Are these tests still in progress or just futures @skriss ?

Comment (skriss) [reply cmd/contour/serve_test.go:66]: I didn't want to go too nuts before getting some 👀  on the change; let me see if I can cover a few more of these now, and I'll add an issue link in the TODO for anything I'm not gonna cover.

Comment (stevesloka) [reply cmd/contour/serve_test.go:66]: awesome, sounds good, just wanted to double check. =)

Comment (skriss) [reply cmd/contour/serve_test.go:66]: Heh I actually just found a bug in upstream Contour by adding some more test cases here, PR incoming

Comment (skriss) [reply cmd/contour/serve_test.go]: OK, added a couple more cases and added an issue link for the TODO.

Comment (skriss) [reply cmd/contour/serve_test.go]: @stevesloka LMK if you had any other comments, otherwise this should be good to merge.

**Comment by sunjayBhatia** [comment]
Coverage numbers going 🆙  😍

**Comment by skriss** [comment]
Eh, guess we can hold this one for 1.15 since it's just a refactor and that'll give it a little more time to bake in dev testing.

**Review by stevesloka — APPROVED** [review]
/lgtm

**Comment by stevesloka** [comment]
@skriss looks like you need a rebase.

**Comment by skriss** [comment]
> @skriss looks like you need a rebase.

Thanks, done!

> @skriss looks like you need a rebase.

**Pull request merged**
